compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -462,6 +462,7 @@ impl<'a, 'tcx> HashStable<StableHashingContext<'a>> for TyS<'tcx> {
 }
 
 #[rustc_diagnostic_item = "Ty"]
+#[pass_by_value]
```

Comment:
You're build issue most certainly comes the beta compiler (which is used when bootstrapping the compiler) which obviously doesn't understand the new attribute (you're trying to add it, previous rustc version cannot know about it).

The fix is to not apply the attribute when bootstrapping, like this:
```suggestion
#[cfg_attr(not(bootstrap), pass_by_value)]
```
PS: i'm just a guy who something hack the compiler and the standard library, not a reviewer by any means

Reply:
@Urgau thanks so much! I had suspected that's the cause but didn't know straight off what would the solution be, will try this! 🙇